packages/instrumentation-socket.io/src/types.ts

```diff
@@ -14,12 +17,22 @@ export interface SocketIoHookInfo {
 export interface SocketIoHookFunction {
     (span: Span, hookInfo: SocketIoHookInfo): void;
 }
+
+export interface HttpTransportInstrumentationConfig {
+    /** Set to the instance of `HttpInstrumentation` used for http instrumentation */
+    httpInstrumentation: HttpInstrumentation;
+    /** Set to the path of socket.io endpoint Desalts to `/socket.io/` */
+    socketPath?: string;
+}
 export interface SocketIoInstrumentationConfig extends InstrumentationConfig {
     /** Hook for adding custom attributes before socket.io emits the event */
     emitHook?: SocketIoHookFunction;
     /** Hook for adding custom attributes before the event listener (callback) is invoked */
     onHook?: SocketIoHookFunction;
-    /** Set to true if you want to trace socket.io reserved events (see https://socket.io/docs/v4/emit-cheatsheet/#Reserved-events) */
+    /** Set to `true` if you want to trace socket.io reserved events (see https://socket.io/docs/v4/emit-cheatsheet/#Reserved-events) */
     traceReserved?: boolean;
+    /** Set to `TransportInstrumentationConfig` if you want to filter out socket.io HTTP transport  */
+    filterHttpTransport?: HttpTransportInstrumentationConfig;
 }
 export type Io = typeof io;
+export type { HttpInstrumentationConfig } from '@opentelemetry/instrumentation-http';
```

Comment:
why export it here?

Reply:
But then it will also be exported from the index of socket.io which is weird IMO.
I think it will be best to import it again in `socket.ts` to use the type